src/encoded/static/components/filegallery.js

```diff
@@ -1134,33 +1134,83 @@ VisualizationControls.defaultProps = {
 // Displays the file filtering controls for the file association graph and file tables.
 
 class FilterControls extends React.Component {
-    constructor() {
-        super();
+    constructor(props, reactContext) {
+        super(props);
+
+        this.navigate = reactContext.navigate;
 
         // Bind this to non-React methods.
         this.handleAssemblyAnnotationChange = this.handleAssemblyAnnotationChange.bind(this);
+        this.setDownloadModalVisibility = this.setDownloadModalVisibility.bind(this);
+        this.handleDownloadClick = this.handleDownloadClick.bind(this);
+
+        //state
+        this.state = { isBatchDownloadOpen: false };
     }
 
     handleAssemblyAnnotationChange(e) {
         this.props.handleAssemblyAnnotationChange(e.target.value);
     }
 
+    setDownloadModalVisibility(isOpen) {
+        this.setState({ isBatchDownloadOpen: isOpen });
+    }
+
+    handleDownloadClick() {
+        const { context, filters } = this.props;
+        const accession = context.accession;
```

Comment:
This will need to use destructuring to work with ENCD-5661.

Reply:
Done